proxy/errors.go

```diff
@@ -0,0 +1,10 @@
+package proxy
+
+type OpenFaaSError struct {
+	StatusCode int
```

Comment:
Does it make sense to have StatusCode in here as we are returning HttpResponse object ?

Reply:
yes. We can start to add logic onto the error later like `IsRetryable()` or `IsAuthinticationError()` to help people out.